src/plugins/ui_actions/public/cell_actions/components/hover_actions_popover.tsx

```diff
@@ -0,0 +1,162 @@
+ * Side Public License, v 1.
+ */
+
+import { EuiPopover, EuiScreenReaderOnly } from '@elastic/eui';
+
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { css } from '@emotion/react';
+import type { Action } from '../../actions';
+import { ActionItem } from './cell_action_item';
+import { ExtraActionsButton } from './extra_actions_button';
+import { ACTIONS_AREA_LABEL, YOU_ARE_IN_A_DIALOG_CONTAINING_OPTIONS } from './translations';
+import { partitionActions } from '../hooks/actions';
+import { ExtraActionsPopOverWithAnchor } from './extra_actions_popover';
+import { CellActionExecutionContext } from './cell_actions';
+
+/** This class is added to the document body while dragging */
+export const IS_DRAGGING_CLASS_NAME = 'is-dragging';
+
+// Overwrite Popover default minWidth to avoid displaying empty space
+const PANEL_STYLE = { minWidth: `24px` };
+
+const hoverContentWrapperCSS = css`
+  padding: 0 ${euiThemeVars.euiSizeS};
+`;
+
+/**
+ * To avoid expensive changes to the DOM, delay showing the popover menu
+ */
+const HOVER_INTENT_DELAY = 100; // ms
+
+interface Props {
+  children: React.ReactNode;
+  getActions: () => Promise<Action[]>;
+  showMoreActionsFrom: number;
+  actionContext: CellActionExecutionContext;
+  showTooltip: boolean;
+}
+
+export const HoverActionsPopover = React.memo<Props>(
+  ({ children, getActions, showMoreActionsFrom, actionContext, showTooltip }) => {
+    const contentRef = useRef<HTMLDivElement>(null);
+    const [isExtraActionsPopoverOpen, setIsExtraActionsPopoverOpen] = useState(false);
+    const [showHoverContent, setShowHoverContent] = useState(false);
+    const [, setHoverTimeout] = useState<number | undefined>(undefined);
+    const popoverRef = useRef<EuiPopover>(null);
+    const [actions, setActions] = useState<Action[] | null>(null);
+
+    const { visibleActions, extraActions } = useMemo(
+      () => partitionActions(actions ?? [], showMoreActionsFrom),
+      [actions, showMoreActionsFrom]
+    );
+
+    const closePopover = useCallback(() => {
+      setHoverTimeout((prevHoverTimeout) => {
+        clearTimeout(prevHoverTimeout);
+        return undefined;
+      });
+      setShowHoverContent(false);
+    }, []);
+
+    const closeExtraActions = useCallback(
+      () => setIsExtraActionsPopoverOpen(false),
+      [setIsExtraActionsPopoverOpen]
+    );
+
+    const onShowExtraActionsClick = useCallback(() => {
+      setIsExtraActionsPopoverOpen(true);
+      closePopover();
+    }, [closePopover, setIsExtraActionsPopoverOpen]);
+
+    const onMouseEnter = useCallback(async () => {
+      // Do not open actions with extra action popover is open
+      if (isExtraActionsPopoverOpen) return;
+
+      // memoize actions after the first call
+      if (actions === null) {
+        getActions().then((newActions) => setActions(newActions));
+      }
```

Comment:
We could use `useAsync` in the parent component, as you did in the inline component, that part looks awesome, the lib already takes care of the unmount thing.
I mean, moving the `useAsync` from the `InlineActions` component to the `CellActions`, and expecting an `actions` array by props instead of the `getActions` async function in the children.
Does it make sense?

Reply:
Hey Sergi! Thanks for reviewing the code and nice catch!
I refactored the `getActions` call to use `useAsyncFn`, which works exactly like `useAsync` but allows us to control when the function is called.

#### About `getActions()` x `actions[]` property:
I decided to pass `getActions()` instead of `actions[]` down the component tree for performance improvement. Because the `HoverActionsPopover` component doesn't show the actions until the component gets hovered, we can delay the `getActions` call to when it needs to be rendered instead of calling it once for every rendered `CellActions` on the page. As you identified, it comes with a cost. We need to deal with async code on both children, but I think it is a price worth paying.

Reply:
Okay I see, makes sense from the processing point of view. However, there is another implication from the UX perspective: we call the `getActions` async function when the users hover, so they may notice a delay if some action takes some time to resolve.

Anyway, I don't think it is a big deal, even though the `isCompatible` function is async, most actions only do synchronous checks. On top of that, I see there's an intentional delay before opening the popover, so we should be good. I think this strategy is better than pre-calling `getActions` for every cell. 👍 

We can tweak the component in the future if we have actions with slow `isCompatible` process time.

I would only like to encapsulate this logic in a `const [actions, loadActions] = useLoadActions()` hook, which will use the `useAsyncFn` internally. This way it will be easier for the different components to use it. 
Also, there's no need to pass the `getActions` by props though all the components, it can be taken from the context.
About the check and the sorting of the `getCompatibleActions` we are doing in the `CellActions`, we can move it to the `CellActionsContextProvider`, so we store the `getActions` in the context `value` directly.
What do you think?